app/src/main/java/nl/eduvpn/app/service/SecurityService.kt

```diff
@@ -137,7 +137,6 @@ class SecurityService(private val context: Context) {
 
         val algorithm =
             signatureString.substring(0, MINISIGN_ALGO_DESCRIPTION_LEGACY.length)
-        //todo: an attacker can decide what algorithm is used, because we do not verify the global signature
```

Comment:
By default the trusted comment does contain "hashed" which indicates the algorithm. It is thus possible to specify the algorithm in the trusted comment, but this would require us to always put the algorithm in the trusted comment.

Reply:
Solutions seems to be to deprecate non prehashed signatures: https://github.com/jedisct1/minisign/issues/104#issuecomment-938591039

Reply:
Yes. The plan is to switch to prehashed, but then all clients need to be updated first, or we'd have to write two different signature files...

Reply:
Also, we have the plan to start using a shared component written in Go thatwould be used by all clients: https://github.com/stevenwdv/eduvpn-common

See wrappers/java